CONTRIBUTING.md

```diff
@@ -1,123 +1,126 @@
+- If you have never contributed to nearcore before and are looking for
+relatively simple issues to get started with, please checkout issues labeled
+with `C-good-first-issue`
+[here](https://github.com/near/nearcore/labels/C-good-first-issue).  If you see
+one that looks interesting and is not claimed, please comment on the issue that
+you would like to start working on it and someone from the team should assign it
+to you.
 
-The PRs should always be against the `master` branch.
+- If you have an idea for an enhancement to the protocol, please make the
+proposal by following the [NEAR Enhancement
+Proposal](https://github.com/near/NEPs/blob/master/neps/nep-0001.md).
 
-The exact process depends on the particular contribution you are making.
+- Otherwise, if you have an idea for an enhancement that does not require a
+change in the proposal, please start by creaeting a new issue in the repository.
+Someone should respond within 48 hours.  If the proposal is accepted, then you
+can follow the process below to begin working on it.
 
-## Typos or small fixes
+- Finally, if you have noticed an obvious typo or bug that can be fixed with a
+small code change, please feel free to submit a PR directly addressing the issue
+without opening an issue.
 
-If you see an obvious typo, or an obvious bug that can be fixed with a small change, in the code or documentation, feel free to submit the pull request that fixes it without opening an issue.
+If you have an issue assigned to you, then we kindly ask that you provide an
+update at least once a week so we know if you are still actively working on the
+issue.
 
-## Working on current tasks
+# Setting up editors and IDEs
 
-If you have never contributed to nearcore before, take a look at the work items in the issue tracker labeled with `C-good-first-issue` [here](https://github.com/near/nearcore/labels/C-good-first-issue). If you see one that looks interesting, and is not claimed, please comment on the issue that you would like to start working on it, and someone from the team will assign it to you.
+Majority of NEAR developers use CLion with Rust plugin as their primary IDE.
 
-Keep in mind the following:
+We also had success with VSCode with rust-analyzer, see the steps for
+installation
+[here](https://commonwealth.im/near/proposal/discussion/338-remote-development-with-vscode-and-rustanalyzer).
 
-1. The changes need to be thoroughly tested. Refer to [this document](https://github.com/nearprotocol/nearcore/wiki/Writing-tests-for-nearcore) for our testing guidelines and overview of the testing infrastructure.
-2. If you get an issue assigned to you, please post updates at least once a week. It is also preferred for you to send a draft PR as early as you have something working, before it is ready.
+Some of us use VIM with [rust.vim](https://github.com/rust-lang/rust.vim) and
```

Comment:
I feel attacked by Emacs not being mentioned. ;)

Reply:
Yeah... I wouldn't be against dropping this section altogether. We basically use "standard rust tooling", it's not helpful to belabor that. It also can be misleading. For vim, for example, I think the current best source of docs is https://sharksforarms.dev/posts/neovim-rust/ (and, for Emach, https://robert.kra.hn/posts/rust-emacs-setup/, although even that doesn't cover eglot)

Reply:
Agreed, removed the whole section.